scripts/lib/html_normalizer.py

```diff
@@ -0,0 +1,52 @@
+import re
+import urllib.parse as url_parser
+
+FOR_REMOVING = [
+    "</?html>",
+    "<head>.*</head>",
+    "</?body>",
+    '<a id="more"></a>',
+]
+
+
+def remove_tags(html: str) -> str:
+    for regex in FOR_REMOVING:
+        html = re.sub(regex, "", html)
+
+    return html
+
+
+def rel_to_abs_url(link: str, base_url: str, rel: str) -> str:
+    abs_url = url_parser.urljoin(base_url, rel)
+    return link.replace(rel, abs_url)
+
+
+def abs_urls(html: str, html_url: str) -> str:
+    return re.sub(
+        r"""(href|src)=["']([^#][^'"]+?)['"]""",
+        lambda x: rel_to_abs_url(x[0], html_url, x[2]),
+        html,
+    )
+
+
+def url_escape(m: re.Match) -> str:
+    return re.sub(r"&(?=[\w\d]+?=)", "&amp;", m[0])
+
+
+def xml_urls(html: str) -> str:
+    return re.sub(r"""<a.+?href=["'][^'"]+?['"]""", url_escape, html)
+
+
+def anchors(html: str) -> str:
+    return re.sub(
+        r"""<a.+?(id=["'][^'"]+?['"]).*?>""", lambda x: f"{x[0]}<span {x[1]}/>", html,
+    )
+
+
+def normalize(html: str, url: str) -> str:
```

Comment:
Not sure is it good place, but perhaps here we also can do some fixes, fro example I checked some broken links and looks like at least some of them can be easily fixed on our side:
```
# not needed dash after "beliefs"
http://www.overcomingbias.com/2007/07/making-beliefs-.html

# not needed `em` tag
https://www.<em>lessw</em>rong.com/posts/4Bwr6s9dofvqPWakn/science-as-attire
```
I think we will have a lot of such sort of fixes